### DISCUSSION TITLE
Network interface

### DISCUSSION THREAD

**Discussion by hueythecat** [dtcooper/raspotify#558] [General]
Is there a way to specify which interface Raspotify advertises on? My device has a hardwired connection thats on an alternate network to that of the wifi. Whenever its up the I assume raspotify favours it & I can only resolve by disabling the wired interface and restarting the raspotify service so it uses the wifi interface.

**Comment by JasonLG1979** [comment]
I have not tried it personally but looking at the systemd docs you could create a drop in override and use the RestrictNetworkInterfaces option.

**Reply by JasonLG1979** [reply]
Well scratch that. The version of systemd in Debian stable is not new enough to support that feature from what I can tell. I just tried it and it doesn't work.

**Reply by JasonLG1979** [reply]
But to answer your question more directly no there is no option in librespot to restrict which interface it uses. That's a little out of scope IMHO. You'd use lower level mechanisms like the above systemd magic (if it actually worked) or some other traffic routing mechanism. If you really want fine grained control like that you should maybe start thinking about containerization.
